src/cplscheme/SerialCouplingScheme.cpp

```diff
@@ -21,321 +21,131 @@ SerialCouplingScheme::SerialCouplingScheme(
     CouplingMode                  cplMode,
     int                           maxIterations)
     : BaseCouplingScheme(maxTime, maxTimeWindows, timeWindowSize, validDigits, firstParticipant,
-                         secondParticipant, localParticipant, m2n, maxIterations, dtMethod)
-{
-  _couplingMode = cplMode;
-  // Coupling mode must be either Explicit or Implicit when using SerialCouplingScheme.
-  PRECICE_ASSERT(_couplingMode != Undefined);
-  if (_couplingMode == Explicit) {
-    PRECICE_ASSERT(maxIterations == 1);
+                         secondParticipant, localParticipant, m2n, maxIterations, cplMode, dtMethod) {
+  if (dtMethod == constants::FIRST_PARTICIPANT_SETS_TIME_WINDOW_SIZE) {
+    if (doesFirstStep()) {
+      _participantSetsTimeWindowSize = true;
+      setTimeWindowSize(UNDEFINED_TIME_WINDOW_SIZE);
+    } else {
+      _participantReceivesTimeWindowSize = true;
+    }
   }
 }
 
-void SerialCouplingScheme::initialize(
-    double startTime,
-    int    startTimeWindow)
+void SerialCouplingScheme::receiveAndSetTimeWindowSize()
 {
-  PRECICE_TRACE(startTime, startTimeWindow);
-  PRECICE_ASSERT(not isInitialized());
-  PRECICE_ASSERT(math::greaterEquals(startTime, 0.0), startTime);
-  PRECICE_ASSERT(startTimeWindow >= 0, startTimeWindow);
-  setTime(startTime);
-  setTimeWindows(startTimeWindow);
+  PRECICE_TRACE();
+  if (_participantReceivesTimeWindowSize) {
+    double dt = UNDEFINED_TIME_WINDOW_SIZE;
+    getM2N()->receive(dt);
+    PRECICE_DEBUG("Received time window size of " << dt << ".");
+    PRECICE_ASSERT(not math::equals(dt, UNDEFINED_TIME_WINDOW_SIZE));
+    PRECICE_ASSERT(not doesFirstStep(), "Only second participant can receive time window size.");
+    setTimeWindowSize(dt);
+  }
+}
 
-  if (_couplingMode == Implicit) {
+void SerialCouplingScheme::checkConfiguration()
+{
+  if (isImplicitCouplingScheme()) {
     PRECICE_CHECK(not getSendData().empty(), "No send data configured! Use explicit scheme for one-way coupling.");
-    if (not doesFirstStep()) {
-      if (not _convergenceMeasures.empty()) {
-        setupConvergenceMeasures();       // needs _couplingData configured
-        setupDataMatrices(getSendData()); // Reserve memory and initialize data with zero
-      }
-      if (getAcceleration().get() != nullptr) {
-        getAcceleration()->initialize(getSendData()); // Reserve memory, initialize
-      }
-    } else if (getAcceleration().get() != nullptr && getAcceleration()->getDataIDs().size() > 0) {
+    if (doesFirstStep() && getAcceleration() && not getAcceleration()->getDataIDs().empty()) {
```

Comment:
This check already existed before. See https://github.com/precice/precice/blob/1d2995325b6e5d21d999e9c2287457721526a3da/src/cplscheme/SerialCouplingScheme.cpp#L55-L60

I tried out putting a `PRECICE_ASSERT(0)` inside the `if`-branch:
```
if (doesFirstStep() && getAcceleration() && not getAcceleration()->getDataIDs().empty()) {
  PRECICE_ASSERT(0);
}
```
The assertion fired, so obviously the branch is reached.

@uekerman If I understand you correctly, this should not be the case.

Reply:
Maybe the acceleration is added to both participants in the configuration. We sometimes do this though it should not be needed. Still, this branch makes no sense, right?

Reply:
> We sometimes do this though it should not be needed.

Should we try to find out where it happens and clean it up? Now would be a good possibility. I'm not sure whether this PR is the right place, though.

> Still, this branch makes no sense, right?

I tried to configure preCICE wrongly such that this `PRECICE_CHECK` fires and this happens, indeed. I used the cpp solverdummy with the following (invalid?) configuration:

```
    <coupling-scheme:serial-implicit> 
      <participants first="SolverOne" second="SolverTwo" /> 
      <max-time-windows value="10" />
      <time-window-size value="1.0" />
      <max-iterations value="3" />
      <min-iteration-convergence-measure min-iterations="5" data="Forces" mesh="MeshOne"/>
      <exchange data="Forces"     mesh="MeshOne" from="SolverOne" to="SolverTwo" />
      <exchange data="Velocities" mesh="MeshOne" from="SolverTwo" to="SolverOne"/>
      <acceleration:IQN-ILS> 
         <data name="Forces" mesh="MeshOne"/>
         <initial-relaxation value="0.01"/>
         <max-used-iterations value="3"/>
         <time-windows-reused value="1"/>
         <filter type="QR2" limit="1e-3"/>
      </acceleration:IQN-ILS>
    </coupling-scheme:serial-implicit>
```

Here, the check is triggered. If I apply the acceleration on the `Velocities`, everything is fine:

```
    <coupling-scheme:serial-implicit> 
      <participants first="SolverOne" second="SolverTwo" /> 
      <max-time-windows value="10" />
      <time-window-size value="1.0" />
      <max-iterations value="3" />
      <min-iteration-convergence-measure min-iterations="5" data="Forces" mesh="MeshOne"/>
      <exchange data="Forces"     mesh="MeshOne" from="SolverOne" to="SolverTwo" />
      <exchange data="Velocities" mesh="MeshOne" from="SolverTwo" to="SolverOne"/>
      <acceleration:IQN-ILS> 
         <data name="Velocities" mesh="MeshOne"/>
         <initial-relaxation value="0.01"/>
         <max-used-iterations value="3"/>
         <time-windows-reused value="1"/>
         <filter type="QR2" limit="1e-3"/>
      </acceleration:IQN-ILS>
    </coupling-scheme:serial-implicit> 
```

This means that we should keep ``PRECICE_CHECK`, since otherwise the user might configure preCICE wrongly. However, this check should be moved into `CouplingSchemeConfiguration`.

For this PR I would suggest to ignore this branch and the `PRECICE_CHECK`, since https://github.com/precice/precice/pull/729 is dealing with it anyways.